app/controllers/admin_controller.rb

```diff
@@ -0,0 +1,35 @@
+class AdminController < ApplicationController
+  include Pundit
+
+  layout "admin"
+
+  protect_from_forgery with: :exception
+
+  before_action :require_login
+  before_action :require_admin
+
+  protected
+
+  def require_login
+    redirect_to login_path unless signed_in?
+  end
+
+  def require_admin
+    redirect_to root_path unless current_user.role? :admin
```

Comment:
This can be somewhat risky behavior because it acknowledges that the page exists but that the current user doesn't have access to it, whereas a 404 wouldn't reveal it's existence at all.